src/GeekLearning.Email/IEmailAddress.cs

```diff
@@ -1,9 +1,13 @@
-﻿namespace GeekLearning.Email
+namespace GeekLearning.Email
 {
+    public enum AddressTarget {  From, To, Cc, Bcc, ReplyTo}
+    
     public interface IEmailAddress
     {
         string Email { get; }
 
         string DisplayName { get; }
+        
+        AddressTarget AddressAs { get;  } 
```

Comment:
`IEmailAddress` is meant to be a simple interface consumer can implement on their on types (or through a facade) so they can use such objects directly. 

I think `AddressTarget` should be an argument for `IEmailSender` methods overloads.

Reply:
I understand your idea about IEmailAddress, just defining the conventional "name <user@domain.tld>" definition. So, what might be an elegant way to link the email user with their 'AddressTarget' attributes? I think I was guessing your thinking when I made the AddressAs property optional. It can be left out, and will default to a 'To:' address insertion. But to avoid the original params implementation (which gets awkward if you want to define 5-10 or more recipients!), and keep the original simplicity of the IEmailAddress contract, maybe there has to have as inputs some kind of User structure, in which the IEmailAddress is itself a property? or overload the class methods? It would be nice to avoid too much of that.